dlib/dnn/core.h

```diff
@@ -3273,8 +3273,8 @@ namespace dlib
                 // layer.
                 const long num_samples = rnd.get_random_32bit_number()%4+3;
                 const long k  = rnd.get_random_32bit_number()%4+2;
-                const long nr = rnd.get_random_32bit_number()%4+2;
-                const long nc = rnd.get_random_32bit_number()%4+2;
+                const long nr = ((rnd.get_random_32bit_number()%4)/2)*2+2;
+                const long nc = ((rnd.get_random_32bit_number()%4)/2)*2+2;
```

Comment:
I am not sure about this. I had to add this change to make sure the number of rows and columns in the tensor is even, so that the test with `reorg_<2, 2>` passes… We can remove this change and the test, if you don't like it.

Reply:
Na, it's fine.